ale/transformation.py

```diff
@@ -1,194 +1,99 @@
 import numpy as np
+import networkx as nx
+from networkx.algorithms.shortest_paths.generic import shortest_path
+
+import spiceypy as spice
 
 from ale.rotation import ConstantRotation, TimeDependentRotation
 
-class FrameNode():
+class FrameChain(nx.DiGraph):
     """
-    A single frame in a frame tree. This class is largely adapted from the Node
-    class in the vispy scene graph implementation.
+    This class is responsible for handling rotations between reference frames.
+    Every node is the reference frame and every edge represents the rotation to
+    between those two nodes. Each edge is directional, where the source --> destination
+    is one rotation and destination --> source is the inverse of that rotation.
 
     Attributes
     __________
-    id : int
-         The NAIF ID code for the frame
-    parent : FrameNode
-             The parent node in the frame tree
-    children : List of FrameNode
-               The children nodes in the frame tree
-    rotation : ConstantRotation or TimeDependentRotation
-               The rotation from this frame to the frame of the parent node
+    frame_changes : list
+                    A list of tuples that represent the rotation from one frame
+                    to another. These tuples should all be NAIF codes for
+                    reference frames
+    ephemeris_time : list
+                     A of ephemeris times that need to be rotated for each set
+                     of frame rotations in the frame chain
```

Comment:
For right now we are storing all of our time dependent rotations at the same time, but in the future we need to go away from this. The current ISIS functionality is to store each time dependent rotation at different times based on what information is available in the SPICE kernels and reduction within a tolerance.